### PR TITLE
feat - Video listing

### PR DESCRIPTION
**Pull request by arpit01923**
Description-

- videoListing page should be added
- contains a api data
- add components navbar, main, footer
- add filtered part

File to check-

- src/contexts/video-listing.jsx
- src/pages/video-listing/main.jsx
- src/reducer/video-listing.jsx
- src/utils/video-listing.jsx

Preview-

https://user-images.githubusercontent.com/86866251/164774410-d7a86336-dd95-41f8-86e8-8d3e7b4be632.mp4

### CODE REVIEW

**Comment by govardhanreddyc79** [comment]
Clean and readable code.
Good folder structure

File: src/pages/video-listing/main.jsx [hunk 1]
```diff
@@ -2,28 +2,51 @@ import { videos } from "backend/db/videos";
...
+              onClick={() =>
+                videoDispatch({ type: "category", payload: "all" })
+              }
+            >
+              All
+            </Link>
+            {home.data.map(({ title }) => (
+              <Link
+                to={`/videoListing/${title}`}
+                style={{ cursor: "pointer" }}
```
Comment (bajajcodes) [inline src/pages/video-listing/main.jsx]: This can be done in css files imported.

File: src/pages/video-listing/video-listing.jsx [hunk 2]
```diff
@@ -1,7 +1,44 @@
...
+      }
+      if (category === "Harsh Gujral") {
+        videoDispatch({ type: "category", payload: "Harsh Gujral" });
+      }
+    }
+  }
+
+  useEffect(() =>{
+    document.title = "VideoListing | Laugh Factory";
+    getVideoListing(category, videoDispatch);
```
Comment (bajajcodes) [inline src/pages/video-listing/video-listing.jsx:34]: This line of code works?

File: src/utils/video-listing.jsx [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const getFilteredVideos = ([...videos], state) => {
```
Comment (bajajcodes) [inline src/utils/video-listing.jsx]: Argument named state, has different descriptive meaning than for what being used here.

**Review by bajajcodes — CHANGES_REQUESTED** [review]

File: src/pages/video-listing/video-listing.jsx (hunk 2 above)
Comment (arpit01923) [reply src/pages/video-listing/video-listing.jsx:34]: yaah works for home category

**Pull request merged**